File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/artifact_list.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ArtifactList(object):
    """
    Result of listing Artifacts.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new ArtifactList object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param root_uri:
            The value to assign to the root_uri property of this ArtifactList.
        :type root_uri: str

        :param files:
            The value to assign to the files property of this ArtifactList.
        :type files: list[oci.aidataplatform_dp.models.ArtifactFileInfo]

        :param next_page_token:
            The value to assign to the next_page_token property of this ArtifactList.
        :type next_page_token: str

        """
        self.swagger_types = {
            'root_uri': 'str',
            'files': 'list[ArtifactFileInfo]',
            'next_page_token': 'str'
        }

        self.attribute_map = {
            'root_uri': 'root_uri',
            'files': 'files',
            'next_page_token': 'next_page_token'
        }

        self._root_uri = None
        self._files = None
        self._next_page_token = None

    @property
    def root_uri(self):
        """
        Gets the root_uri of this ArtifactList.
        Root artifact directory for the run.


        :return: The root_uri of this ArtifactList.
        :rtype: str
        """
        return self._root_uri

    @root_uri.setter
    def root_uri(self, root_uri):
        """
        Sets the root_uri of this ArtifactList.
        Root artifact directory for the run.


        :param root_uri: The root_uri of this ArtifactList.
        :type: str
        """
        self._root_uri = root_uri

    @property
    def files(self):
        """
        Gets the files of this ArtifactList.
        File location and metadata for artifacts.


        :return: The files of this ArtifactList.
        :rtype: list[oci.aidataplatform_dp.models.ArtifactFileInfo]
        """
        return self._files

    @files.setter
    def files(self, files):
        """
        Sets the files of this ArtifactList.
        File location and metadata for artifacts.


        :param files: The files of this ArtifactList.
        :type: list[oci.aidataplatform_dp.models.ArtifactFileInfo]
        """
        self._files = files

    @property
    def next_page_token(self):
        """
        Gets the next_page_token of this ArtifactList.
        Token that can be used to retrieve the next page of artifact results.


        :return: The next_page_token of this ArtifactList.
        :rtype: str
        """
        return self._next_page_token

    @next_page_token.setter
    def next_page_token(self, next_page_token):
        """
        Sets the next_page_token of this ArtifactList.
        Token that can be used to retrieve the next page of artifact results.


        :param next_page_token: The next_page_token of this ArtifactList.
        :type: str
        """
        self._next_page_token = next_page_token

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/ArtifactList.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Result of listing artifacts.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ArtifactList.Builder.class)

public final class ArtifactList  {
    @Deprecated
    @java.beans.ConstructorProperties({"rootUri", "files", "nextPageToken"})
    public ArtifactList(String rootUri, java.util.List<ArtifactFileInfo> files, String nextPageToken) {
        super();
        this.rootUri = rootUri;
        this.files = files;
        this.nextPageToken = nextPageToken;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Root artifact directory for the run.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("root_uri")
private String rootUri;

        /**
         * Root artifact directory for the run.
         * @param rootUri the value to set
         * @return this builder
         **/
        

public Builder rootUri(String rootUri) {
    this.rootUri = rootUri;
    return this;
}
            /**
     * File location and metadata for artifacts.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("files")
private java.util.List<ArtifactFileInfo> files;

        /**
         * File location and metadata for artifacts.
         * @param files the value to set
         * @return this builder
         **/
        

public Builder files(java.util.List<ArtifactFileInfo> files) {
    this.files = files;
    return this;
}
            /**
     * Token that can be used to retrieve the next page of artifact results.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("next_page_token")
private String nextPageToken;

        /**
         * Token that can be used to retrieve the next page of artifact results.
         * @param nextPageToken the value to set
         * @return this builder
         **/
        

public Builder nextPageToken(String nextPageToken) {
    this.nextPageToken = nextPageToken;
    return this;
}


        public ArtifactList build() {
            ArtifactList model = new ArtifactList(this.rootUri
                , this.files
                , this.nextPageToken);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ArtifactList model) {
                this.rootUri(model.getRootUri());
    this.files(model.getFiles());
    this.nextPageToken(model.getNextPageToken());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Root artifact directory for the run.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("root_uri")
    private final String rootUri;

        /**
     * Root artifact directory for the run.
     * @return the value
     **/
    
    public String getRootUri() {
        return rootUri;
    }


        /**
     * File location and metadata for artifacts.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("files")
    private final java.util.List<ArtifactFileInfo> files;

        /**
     * File location and metadata for artifacts.
     * @return the value
     **/
    
    public java.util.List<ArtifactFileInfo> getFiles() {
        return files;
    }


        /**
     * Token that can be used to retrieve the next page of artifact results.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("next_page_token")
    private final String nextPageToken;

        /**
     * Token that can be used to retrieve the next page of artifact results.
     * @return the value
     **/
    
    public String getNextPageToken() {
        return nextPageToken;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ArtifactList(");
        sb.append("rootUri=").append(String.valueOf(this.rootUri));
        sb.append(", files=").append(String.valueOf(this.files));
        sb.append(", nextPageToken=").append(String.valueOf(this.nextPageToken));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ArtifactList)) {
            return false;
        }

        ArtifactList other = (ArtifactList) o;
        return java.util.Objects.equals(this.rootUri, other.rootUri) &&
            java.util.Objects.equals(this.files, other.files) &&
            java.util.Objects.equals(this.nextPageToken, other.nextPageToken);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.rootUri == null ? 43 : this.rootUri.hashCode());
        result = (result * PRIME) + (this.files == null ? 43 : this.files.hashCode());
        result = (result * PRIME) + (this.nextPageToken == null ? 43 : this.nextPageToken.hashCode());
        return result;
    }


}
